### PR TITLE
Changeable FileSystem for FileHandler

### PR DESCRIPTION
**Pull request by tchssk**
goa can serve static assets via the DSL `Files` but it always uses the native file system. This patch makes it possible to use another `FileSystem` for that.

### CODE REVIEW

**Comment by raphael** [comment]
Thanks! the patch looks good. Do you think you could add a simple test as well?

**Comment by tchssk** [comment]
I added a test.

File: service_test.go [hunk 1]
```diff
@@ -307,6 +309,65 @@ var _ = Describe("Service", func() {
...
+			ctrl := s.NewController("test")
+			handler = ctrl.FileHandler("/swagger.json", "public/swagger/swagger.json")
+			muxHandler = ctrl.MuxHandler("testAct", handler, nil)
+		})
+
+		AfterEach(func() {
+			os.RemoveAll(outDir)
+		})
+
+		It("creates a handle", func() {
```
Comment (raphael) [inline service_test.go]: handle -> handler

**Comment by raphael** [comment]
Thank you the test looks great! (spotted a typo - left a comment). I guess it doesn't test overriding `FileSystem` but it's OK since the test proves that the default implementation is correctly invoked.

**Comment by tchssk** [comment]
https://github.com/goadesign/goa/pull/1296/commits/4d8adde3da050dd6c01d5bfbc255c6e301e105e0 fixes typos.

**Comment by raphael** [comment]
Thank you!

**Comment by raphael** [comment]
I guess this could be backported to `v1` as well

**Comment by tchssk** [comment]
Thank you! I'll backport this to `v1`.

**Pull request merged**